test/jdk/jdk/crac/newArgs/FailedCheckpointRestoreTest.java

```diff
@@ -0,0 +1,131 @@
+/*
+ * Copyright (c) 2025, Azul Systems, Inc. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Azul Systems, 385 Moffett Park Drive, Suite 115, Sunnyvale
+ * CA 94089 USA or visit www.azul.com if you need additional information or
+ * have any questions.
+ */
+
+import java.util.List;
+
+import jdk.crac.CheckpointException;
+import jdk.crac.Context;
+import jdk.crac.Core;
+import jdk.crac.Resource;
+import jdk.crac.RestoreException;
+import jdk.test.lib.crac.CracBuilder;
+import jdk.test.lib.crac.CracTest;
+import jdk.test.lib.crac.CracTestArg;
+import jdk.test.lib.process.OutputAnalyzer;
+
+/**
+ * @test
+ * @summary New main should run only if C/R succeeds.
+ * @library /test/lib
+ * @build FailedCheckpointRestoreTest
+ * @run driver jdk.test.lib.crac.CracTest SUCCESS
+ * @run driver jdk.test.lib.crac.CracTest CHECKPOINT_EXCEPTION
+ * @run driver jdk.test.lib.crac.CracTest RESTORE_EXCEPTION
+ * @requires (os.family == "linux")
+ */
+public class FailedCheckpointRestoreTest implements CracTest {
+    private static final String NEW_MAIN_CLASS = "FailedCheckpointRestoreTest$InternalMain";
+    private static final String RESTORE_MSG = "RESTORED";
+    private static final String NEW_MAIN_MSG = "Hello from new main!";
+
+    public enum Variant {
+        SUCCESS,
```

Comment:
The `SUCCESS` variant looks identical to `CheckpointInNewMainTest`, except for registering a resource that wouldn't do anything. Could you merge these?